Makefile

```diff
@@ -113,6 +118,21 @@ create-kubeconfig-cm:
 delete-kubeconfig-cm:
 	kubectl -n kube-system delete configmap $(KUBECONFIG_CM)
 
+.PHONY : tls-secret
+tls-secrets: gen-certs
+	# Do clean up first if there's any left-over secrets
+	kubectl -n $(KUBENAMESPACE) delete secret $(K8S_SECRETE_MGR_SERVER) || true
+	kubectl -n $(KUBENAMESPACE) delete secret $(K8S_SECRETE_MGR_CLIENT) || true
+	kubectl -n $(KUBENAMESPACE) delete secret $(K8S_SECRETE_AGENT_CLIENT) || true
+	# Create new secrets from generated certs
+	kubectl -n $(KUBENAMESPACE) create secret generic $(K8S_SECRETE_MGR_SERVER) --from-file=./hack/tls/certs/inframanager/server/
+	kubectl -n $(KUBENAMESPACE) create secret generic $(K8S_SECRETE_MGR_CLIENT) --from-file=./hack/tls/certs/inframanager/client/
+	kubectl -n $(KUBENAMESPACE) create secret generic $(K8S_SECRETE_AGENT_CLIENT) --from-file=./hack/tls/certs/infraagent/client/
+
+.PHONY : gen-certs
+gen-certs:
+	 ./hack/tls/gen_certs.sh
```

Comment:
We should keep the generation of keys and certificates as decoupled from k8s as possible. For that, the makefile should not be invoking scripts internally to automatically generated these files. Instead, the makefile should assume that certificates have been pre-created at the required location and so, it can go and grab them for usage. The script can be provided purely as a standalone reference so that user can use it or have their own way of generating these files with their choice of encryption algorithm and keys/certificates, etc. as we discussed in the context of SDLe.

Reply:
The secrets with certs & keys are still optional and a user can always provide their own files and create the K8s secrets manually if they wish. These make targets are not pre-requisites for deploy but the secrets are. It does not matter if they create them with provide make target or manually.

Reply:
moved gen-certs under scripts dir.